server/market/market.go

```diff
@@ -95,11 +91,79 @@ type Market struct {
 	storage db.DEXArchivist
 }
 
+// NewMarket creates a new Market for the provided base and quote assets, with
+// an epoch cycling at given duration in milliseconds.
+func NewMarket(mktInfo *dex.MarketInfo, storage db.DEXArchivist, swapper Swapper, authMgr AuthManager,
+	coinLockerBase, coinLockerQuote coinlock.CoinLocker) (*Market, error) {
+	// Make sure the DEXArchivist is healthy before taking orders.
+	if err := storage.LastErr(); err != nil {
+		return nil, err
+	}
+
+	// Lock coins backing active orders.
+	baseCoins, quoteCoins, err := storage.ActiveOrderCoins(mktInfo.Base, mktInfo.Quote)
+	if err != nil {
+		return nil, err
+	}
+	// TODO: first check that the coins aren't already locked by e.g. another market.
+	coinLockerBase.LockCoins(baseCoins)
+	coinLockerQuote.LockCoins(quoteCoins)
+
+	return &Market{
+		running:         make(chan struct{}),
+		marketInfo:      mktInfo,
+		orderRouter:     make(chan *orderUpdateSignal, 16),
+		book:            book.New(mktInfo.LotSize),
+		matcher:         matcher.New(),
+		epochOrders:     make(map[order.OrderID]order.Order),
+		swapper:         swapper,
+		auth:            authMgr,
+		storage:         storage,
+		coinLockerBase:  coinLockerBase,
+		coinLockerQuote: coinLockerQuote,
+	}, nil
+}
+
+// SetStartEpochIdx sets the starting epoch index. This should generally be
+// called before Run, or Start used to specify the index at the same time.
+func (m *Market) SetStartEpochIdx(startEpochIdx int64) {
+	atomic.StoreInt64(&m.startEpochIdx, startEpochIdx)
+}
+
+// StartEpochIdx gets the starting epoch index.
+func (m *Market) StartEpochIdx() int64 {
+	return atomic.LoadInt64(&m.startEpochIdx)
+}
+
+// Start beings order processing with a starting epoch index. See also
+// SetStartEpochIdx and Run. Stop the Market by cancelling the context.
+func (m *Market) Start(ctx context.Context, startEpochIdx int64) {
+	m.SetStartEpochIdx(startEpochIdx)
+	m.Run(ctx)
+}
+
+// WaitForEpochOpen waits until the start of epoch processing.
+func (m *Market) WaitForEpochOpen() {
+	<-m.running
+}
```

Comment:
My initial thought was it would be helpful to delay the startup of other subsystems such as the comms.Server or perhaps the BookRouter, but really the OrderRouter will get error messages back from the Market's epoch processing loop that would be just as good as what the OrderRouter would provide to the client if it had known the market(s) were not yet running before submitting the orders for processing.
Delaying the start of the Server might be helpful eventually but I also don't think that's critical.

unexporting as suggested.